### ISSUE TITLE
Using this as a plugin in Nuxt 3 breaks HMR

### ISSUE THREAD

**Issue by pnwnelson** [victorybiz/vue-simple-acl#15]
Would love to use it in a Nuxt 3 application but can't when the HMR is broken and really slows me down. 
There are no errors, but seeing my DOM changes requires a page reload.

**Comment by zenojunior** [comment]
Using with Vite, breaks too

**Comment by Nymrinae** [comment]
> Using with Vite, breaks too

Any solution so far?

**Comment by clabnet** [comment]
Also for me, same issue using Vite Vue3. 
HMR is not available when using this as plugin.
[Here](https://github.com/clabnet/vite-naive-uno-acl-breaks-hmr) a repo showcase

I have also created a new npm package [@clabnet/vue-simple-acl](https://www.npmjs.com/package/@clabnet/vue-simple-acl)
with all packages updated to vite 4.x, but the problem is the same, HMR is breaks.

@pnwnelson  @zenojunior  @Nymrinae Have you notice ? The author @victorybiz seems do not respond a long time ago.

**Comment by Nymrinae** [comment]
> Have you notice ? The author @victorybiz seems do not respond a long time ago.

Yeah, the fact that it breaks HMR is a huge issue for me, I consider swapping to CASL instead but didn't try yet if it doesn't break it as well, I hope not.

**Comment by clabnet** [comment]
I have fixed my new npm package [@clabnet/vue-simple-acl](https://www.npmjs.com/package/@clabnet/vue-simple-acl) with HMR running fine. I had to completely override the build mode, it was tough but it seems to work. Try it if you can and give me your feedback, please.

**Comment by Nymrinae** [comment]
> I have fixed my new npm package [@clabnet/vue-simple-acl](https://www.npmjs.com/package/@clabnet/vue-simple-acl) with HMR running fine. I had to completely override the build mode, it was tough but it seems to work. Try it if you can and give me your feedback, please.

Works fine for me, thanks a lot!!

**Comment by Tob0t** [comment]
Thanks @clabnet for your package, I also switched to your library now due to missing HMR.
Could you maybe also push your changes to github so we know which part of the source code differs from the actual package?

**Comment by psarin** [comment]
Same issure here. However, in using @clabnet version, I get an error saying Module not found: Error: Can't resolve './utils' in '/node_modules/@clabnet/vue-simple-acl/dist' because  origin is strict.
